source/particle/interpolator/cell_average.cc

```diff
@@ -74,8 +75,10 @@ namespace aspect
                 ghost_particles.equal_range(cell_index);
 
         const unsigned int n_particles = std::distance(particle_range.first,particle_range.second);
-        const unsigned int n_properties = particles.begin()->second.get_properties().size();
-        std::vector<double> cell_properties (n_properties,0.0);
+        const unsigned int n_particle_properties = particles.begin()->second.get_properties().size();
+        const unsigned int n_selected_properties = selected_properties.n_selected_components(n_particle_properties);
+
+        std::vector<double> cell_properties (n_selected_properties,0.0);
```

Comment:
Thanks. We had many places like this in deal.II in the early years where you needed to translate, and it was (i) a constant hassle to get the translation correct in both direction because you needed additional information that may not always have been easy to compute, and (ii) it is so much hard to explain in the documentation how exactly the translation happens than just to say "fill the indicated locations and don't touch the rest".